### DISCUSSION TITLE
merging multiple ArchR objects

### DISCUSSION THREAD

**Discussion by rojinsafavi** [GreenleafLab/ArchR#768] [Questions / Documentation]
Is it possible to merge/combine multiple ArchR objects?
To give more context, I have in-vivo and in-vitro dataset. I would like to perform batch correction on each independently, and they combine them

**Comment by rcorces** [comment]
This is still not possible unfortunately.

**Reply by rojinsafavi** [reply]
@rcorces Would it be possible to convert Signac obj to ArchR? I'm thinking of creating ArchR obj independently, convert each to Signac, combine the vivo and vitro in Signac, and hopefully be able to come back to ArchR from Signac obj

**Reply by rcorces** [reply]
Interconversion of objects from Signac is not currently possible. Maybe your application doesnt really require all of this? You can manually add or update dimensionality reductions and embeddings in ArchR. Perhaps that is all you would need to do?

**Reply by rojinsafavi** [reply]
I will also need the GIMs, would you please share the R command for manual updates?

**Reply by rcorces** [reply]
I'm not sure what "GIMs" means.

Here is some example code for how to manually add an embedding. The same principal should apply to any object that is just stored in the `ArchRProject` (i.e. not stored in arrows). You should just look at what the object looks like in the `ArchRProject` and create it manually. In this example, `colData2` contains a UMAP embedding given by `UMAP1` and `UMAP2` as well as cluster definitions for all of the cells which are loaded into `cellColData`.

```
all.equal(rownames(colData2), proj2$cellNames)
df <- DataFrame(row.names=proj2$cellNames, "custom#UMAP1" = colData2$UMAP1, "custom#UMAP2" =  colData2$UMAP2, check.names = FALSE)
proj2@embeddings$customUMAP <- SimpleList(df = df, params = list())
proj2@cellColData$Clusters <- colData2$Clusters
```

**Comment by Zepeng-Mu** [comment]
I'm trying to do the same thing. Essentially I want to merge cell barcodes and associated data from two filtered ArchR projects, and then re-do dimension reduction and downstream analysis using the combined dataset.

I tried using `ArchRProject()` with the ArrowFiles from the project that I want to merge. In this way I'm hoping to make a new project using filtered separate projects. Is this the right way to do it?

Both of my projects have about 7K cells, and this can be checked when I load the projects separately. But in the project combined from my two ArrowFiles, there are 90K cells. It seems that the cells that are filtered out in previous projects are back in the newly combined project. I can always filter the project again. But I'm wondering whether this is the intended behavior of loading >1 arrowFiles into a project?

**Reply by Zepeng-Mu** [reply]
I now see it may because of `dropCells=F` in `saveArchRProject` function.
